src/main/java/org/fenixedu/academic/domain/ExecutionCourse.java

```diff
@@ -118,11 +118,26 @@ public int compare(ExecutionCourse o1, ExecutionCourse o2) {
 
                 @Override
                 public int compare(ExecutionCourse o1, ExecutionCourse o2) {
-                    final int cep = o1.getExecutionPeriod().compareTo(o2.getExecutionPeriod());
+                    final int cep = EXECUTION_COURSE_EXECUTION_PERIOD_COMPARATOR.compare(o1, o2);
                     if (cep != 0) {
                         return cep;
                     }
-                    final int c = Collator.getInstance().compare(o1.getNome(), o2.getNome());
+                    final int c = EXECUTION_COURSE_NAME_COMPARATOR.compare(o1, o2);
+                    return c == 0 ? DomainObjectUtil.COMPARATOR_BY_ID.compare(o1, o2) : c;
+                }
+
+            };
+
+    public static final Comparator<ExecutionCourse> EXECUTION_COURSE_COMPARATOR_BY_EXECUTION_PERIOD_REVERSED_AND_NAME =
```

Comment:
check if ``Comparator.reversed()`` can be used instead of this comparator.

Reply:
`Comparator.reversed()` can be used but the ExecutionCourse order will also be reversed alphabetically.